### PR TITLE
Fixed #29112 -- Allowed using QuerySet.update() to update specific keys in nested JSONField.

### PR DESCRIPTION
**Pull request by priyanshsaxena**
Specific keys can be updated within a nested JSONField  by specifying
the path and new value as arguments to the update() method, just like
the syntax used in Lookups.

### CODE REVIEW

**Comment by priyanshsaxena** [comment]
Continuing from #9715

**Comment by felixxm** [comment]
@priyanshsaxena Thanks :+1:. Please provide solution for all builtin backends.

**Comment by priyanshsaxena** [comment]
@felixxm reading up on the history in this [ticket](https://code.djangoproject.com/ticket/12990), and related PRs. Will commit another update soon.

File: django/db/models/sql/subqueries.py [hunk 1]
```diff
@@ -82,6 +83,16 @@ def add_update_values(self, values):
...
         for name, val in values.items():
+            if LOOKUP_SEP in name:
+                attributes = name.split(LOOKUP_SEP)
+                name = attributes.pop(0)
+                hierarchy = ','.join(attributes)
+                field = self.get_meta().get_field(name)
+                from django.contrib.postgres.functions import JSONBSet
+                from django.db.models import Value
+                val = JSONBSet(
+                    field.name, Value('{%s}' % hierarchy), Value('"%s"' % val), Value(False)
```
Comment (bpartridge) [inline django/db/models/sql/subqueries.py:94]: Should the val argument should be the following to accommodate primitive values that aren't strings, and that don't necessarily stringify into valid JSON (such as booleans)? Test coverage should likely also be expanded accordingly. We're using the following in our codebase for the third argument to jsonb_set:

`Cast(Value(json.dumps(value)), output_field=JSONField())`

**Comment by felixxm** [comment]
Closing due to inactivity.

**Pull request closed without merging**